packages/framework/src/Form/Admin/Customer/DeliveryAddressFormType.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Form\Admin\Customer;
+
+use Shopsys\FrameworkBundle\Model\Country\CountryFacade;
+use Shopsys\FrameworkBundle\Model\Customer\DeliveryAddressData;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints;
+
+class DeliveryAddressFormType extends AbstractType
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Country\CountryFacade $countryFacade
+     */
+    public function __construct(private readonly CountryFacade $countryFacade)
+    {
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $countries = $this->countryFacade->getAllEnabledOnDomain($options['domain_id']);
+
+        $builder
+            ->add('firstName', TextType::class, [
+                'required' => true,
+                'constraints' => [
+                    new Constraints\NotBlank(['message' => 'Please enter first name']),
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'First name cannot be longer than {{ limit }} characters',
+                    ]),
+                ],
+                'label' => t('First name'),
+            ])
+            ->add('lastName', TextType::class, [
+                'required' => true,
+                'constraints' => [
+                    new Constraints\NotBlank(['message' => 'Please enter last name']),
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'Last name cannot be longer than {{ limit }} characters',
+                    ]),
+                ],
+                'label' => t('Last name'),
+            ])
+            ->add('companyName', TextType::class, [
+                'required' => false,
+                'constraints' => [
+                    new Constraints\Length([
+                        'max' => 100,
+                        'maxMessage' => 'Company name cannot be longer than {{ limit }} characters',
+                    ]),
+                ],
+                'label' => t('Company'),
+            ])
+            ->add('street', TextType::class, [
+                'required' => true,
```

Comment:
If the field is required, it should contain the `NotBlank` constraint, or? 

Now I can send form only with the first and last name filled, resulting in 500 error

Reply:
You are right, fixed in https://github.com/shopsys/shopsys/pull/3261/commits/29085eb666debe2cc0d438180b27ed8be8313696